webapp/channels/src/components/channel_notifications_modal/components/notification_section.tsx

```diff
@@ -1,90 +1,49 @@
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
-import React from 'react';
+import React, {ChangeEvent} from 'react';
 
 import {NotificationSections, NotificationLevels} from 'utils/constants';
 
 import CollapseView from './collapse_view';
 import ExpandView from './expand_view';
 
-export default class NotificationSection extends React.PureComponent {
-    static propTypes = {
-
-        /**
-         * Notification section
-         */
-        section: PropTypes.string.isRequired,
-
-        /**
-         * Expand if true, else collapse the section
-         */
-        expand: PropTypes.bool.isRequired,
-
-        /**
-         * Member's desktop notification level
-         */
-        memberNotificationLevel: PropTypes.string.isRequired,
-
-        /**
-         * Member's desktop_threads notification level
-         */
-        memberThreadsNotificationLevel: PropTypes.string,
-
-        /**
-         * Ignore channel-wide mentions @channel, @here and @all
-         */
-        ignoreChannelMentions: PropTypes.string,
-
-        /**
-         * User's global notification level
-         */
-        globalNotificationLevel: PropTypes.string,
-
-        /**
-         * onChange handles update of desktop notification level
-         */
-        onChange: PropTypes.func.isRequired,
-
-        /**
-         * onChangeThreads handles update of desktop_threads notification level
-         */
-        onChangeThreads: PropTypes.func,
-
-        /**
-         * Submit function to save notification level
-         */
-        onSubmit: PropTypes.func.isRequired,
-
-        /**
-         * Update function to to expand or collapse a section
-         */
-        onUpdateSection: PropTypes.func.isRequired,
-
-        /**
-         * Error string from the server
-         */
-        serverError: PropTypes.string,
-    };
+type Props = {
+    section: string;
+    expand: boolean;
+    memberNotificationLevel: string;
+    memberThreadsNotificationLevel?: string;
+    ignoreChannelMentions?: string;
+    globalNotificationLevel?: string;
+    onChange: (value: string) => void;
+    onChangeThreads?: (value: string) => void | any;
+    onSubmit: () => void;
+    onUpdateSection: (value: string) => void;
+    serverError?: string;
+}
+export default class NotificationSection extends React.PureComponent<Props> {
+    constructor(props: Props) {
+        super(props);
+    }
 
-    handleOnChange = (e) => {
+    handleOnChange = (e: ChangeEvent<HTMLInputElement>) => {
         this.props.onChange(e.target.value);
-    };
+    }
 
-    handleOnChangeThreads = (e) => {
+    handleOnChangeThreads = (e: ChangeEvent<HTMLInputElement>) => {
         const value = e.target.checked ? NotificationLevels.ALL : NotificationLevels.MENTION;
-
-        this.props.onChangeThreads(value);
-    };
+        if (this.props.onChangeThreads) {
+            this.props.onChangeThreads(value);
+        }
```

Comment:
You can avoid additional if statement by optional chaining. it is basically same is checking if it is null/ undefined with an if statement
```suggestion
        this.props.onChangeThreads?.(value);
```